content/techniques/caching.md

```diff
@@ -95,34 +115,32 @@ handleEvent(client, data) {
 }
 ```
 
-> info **Hint** The `@CacheKey()` decorator is imported from `@nestjs/common` package.
-
 However, the additional `@CacheKey()` decorator is required in order to specify a key used to subsequently store and retrieve cached data. Also, please note that you **shouldn't cache everything**. Actions which perform some business operations rather than simply querying the data should never be cached.
 
 Additionally, you may specify a cache expiration time (TTL) by using the `@CacheTTL()` decorator, which will override the global default TTL value.
 
 ```typescript
 @@filename()
-@CacheTTL(10) // seconds
+@CacheTTL(10)
 @UseInterceptors(CacheInterceptor)
+@SubscribeMessage('events')
 handleEvent(client: Client, data: string[]): Observable<string[]> {
   return [];
 }
 @@switch
-@CacheTTL(10) // seconds
+@CacheTTL(10)
 @UseInterceptors(CacheInterceptor)
+@SubscribeMessage('events')
 handleEvent(client, data) {
   return [];
 }
 ```
 
-> info **Hint** The `@CacheTTL()` decorator is imported from `@nestjs/common` package.
-
-The `@CacheTTL()` decorator may be used with or without a corresponding `@CacheKey()` decorator. This allows you to set custom cache expiration times (TTL) on a per-method basis, overriding global defaults whether or not a corresponding `@CacheKey()` is specified.
+> info **Hint** The `@CacheTTL()` decorator may be used with or without a corresponding `@CacheKey()` decorator. 
 
 #### Different stores
 
-This service take advantage of [cache-manager](https://github.com/BryanDonovan/node-cache-manager) under the hood. The `cache-manager` package supports a wide-range of useful stores, for example, [Redis](https://github.com/dabroek/node-cache-manager-redis-store) store. A full list of supported stores is available [here](https://github.com/BryanDonovan/node-cache-manager#store-engines)). To set up the Redis store, simple pass the package together with corresponding options to the `register()` method.
+This service take advantage of [cache-manager](https://github.com/BryanDonovan/node-cache-manager) under the hood. The `cache-manager` package supports a wide-range of useful stores, for example, [Redis](https://github.com/dabroek/node-cache-manager-redis-store) store. A full list of supported stores is available [here](https://github.com/BryanDonovan/node-cache-manager#store-engines). To set up the Redis store, simple pass the package together with corresponding options to the `register()` method.
```

Comment:
takes advantage <-- (not take) (p.s., I know this was the original, but while you're at it :smile)